### PR TITLE
Retry pytest on fail to avoid flaky errors

### PR DESCRIPTION
**Pull request by mwestphal**
A work around for https://github.com/pytest-dev/pytest/issues/11647

### CODE REVIEW

File: .github/actions/wheels/test_with_pytest_retry.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# Run pytest command two times if needed to pass
+max_retry=2
+counter=0
+until pytest -s $1/python/testing 
+do
+   [[ counter -eq $max_retry ]] && echo "Failed!" && exit 1
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply .github/actions/wheels/test_with_pytest_retry.sh:8]: I can easily check for macos.

I have no idea how to recover the version of python unless by checking it manually with a python command.

I could check the exit code but that doesnt seem required to me.

Let me know, but avoiding flaky issues is not a big deal to me.

**Comment by mwestphal** [comment]
Expected CI output can be seen here:

https://github.com/f3d-app/f3d-superbuild/actions/runs/7043560326/job/19169574953?pr=127

File: .github/actions/wheels/test_with_pytest_retry.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+python_version=$(python --version)
+python_version=${python_version#* }
+python_version=${python_version%.*}
+echo "Python version: $python_version"
+
+# Recover system information
+machine=$(uname -s)
+
+# Try once in general
+# Three time with macOS and python 3.12
```
Comment (Meakk) [inline .github/actions/wheels/test_with_pytest_retry.sh:14]: Ten times

**Review by Meakk — APPROVED** [review]
minor comment then gtg

**Pull request merged**